### PR TITLE
add earthlyls configuration

### PR DESCRIPTION
**Pull request by glehmann**
## Describe your changes
add the configuration to download earthlyls binaries from the github release

## Issue ticket number and link
related to https://github.com/neovim/nvim-lspconfig/pull/3111

## Checklist before requesting a review

- [x] I have successfully tested installation of the package.
- [x] I have successfully tested the package after installation.

### CODE REVIEW

**Comment by glehmann** [comment]
I may be able to provide the binaries for other platform, but I can’t find what `target` to use. Could you point me to a list of supported target or to how I can build them?

File: packages/earthlyls/package.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+name: earthlyls
+description: |
+  A fast language server for earthly.
+homepage: https://github.com/glehmann/earthly
```
Comment (williamboman) [inline packages/earthlyls/package.yaml:5]: I believe the intended URL is:
```suggestion
homepage: https://github.com/glehmann/earthlyls
```

File: schemas/enums/language.json [hunk 2]
```diff
@@ -50,6 +50,7 @@
         "Docker",
         "Drools",
         "DTS",
+        "Earthly",
```
Comment (williamboman) [inline schemas/enums/language.json:53]: I'd suggest using "Earthly" as the language entry. While technically not a language, these entries are primarily meant to allow users to discover relevant tooling and I believe Earthly would be the most appropriate one in this case.

File: packages/earthlyls/package.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+licenses:
+  - MIT
+languages:
+  - Earthly
+categories:
+  - LSP
+
+source:
+  # renovate:versioning=loose
```
Comment (williamboman) [inline packages/earthlyls/package.yaml:14]: The project seems to apply semver versioning, in which case we don't need to override the default versioning strategy.
```suggestion
```

**Comment by williamboman** [comment]
> I may be able to provide the binaries for other platforms, but I can’t find what `target` to use. Could you point me to a list of supported target or to how I can build them?

The full list of supported platforms can be found [here](https://github.com/mason-org/mason-registry/blob/main/schemas/enums/platform.json#L5-L35). The primary ones are `darwin_x64`, `darwin_arm64`, `linux_x64`, `linux_arm64` and `win_x64`, so this LGTM already!

File: packages/earthlyls/package.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      file: earthlyls-{{ version }}-linux-arm64.tar.gz
+      bin: earthlyls-{{ version }}-linux-arm64/earthlyls
+    - target: darwin_x64
+      file: earthlyls-{{ version }}-macos-amd64.tar.gz
+      bin: earthlyls-{{ version }}-macos-amd64/earthlyls
+    - target: darwin_arm64
+      file: earthlyls-{{ version }}-macos-arm64.tar.gz
+      bin: earthlyls-{{ version }}-macos-arm64/earthlyls
+    - target: win_x64
+      file: earthlyls-{{ version }}-windows-amd64.tar.gz
```
Comment (williamboman) [inline packages/earthlyls/package.yaml:29]:
```suggestion
      file: earthlyls-{{ version }}-windows-amd64.zip
```

**Review by williamboman — APPROVED** [review]
LGTM! If you'd like to expand the asset list with more platforms we can do so in later PR

**Pull request merged**